pkg/mimirtool/commands/rules.go

```diff
@@ -412,7 +418,45 @@ func (r *RuleCommand) printRules(_ *kingpin.ParseContext) error {
 	}
 
 	p := printer.New(r.DisableColor)
-	return p.PrintRuleGroups(rules)
+
+	if r.OutputDir != "" {
+		log.Infof("Output dir detected writing rules to directory: %s", r.OutputDir)
+		for namespace, rule := range ruleNS {
+			if err = saveNamespaceRuleGroup(namespace, rule, r.OutputDir); err != nil {
+				return err
+			}
+		}
+
+		// Don't print the rule set if we've specified an output directory to save the rule files. It gets too noisy.
+		return nil
+	}
+
+	return p.PrintRuleGroups(ruleNS)
+}
+
+func saveNamespaceRuleGroup(ns string, ruleGroup []rwrulefmt.RuleGroup, dir string) error {
+	baseDir, err := filepath.Abs(dir)
+	if err != nil {
+		return err
+	}
+
+	if regexp.MustCompile(disallowedNamespaceChars).Match([]byte(ns)) {
```

Comment:
nit: I'd define <code>var disallowdNamespaceCharsRegex = regexp.MustCompile(\`[<>:"/\\|?*\x00-\x1F]\`)</code> to 1) compile this early and crash if regex is wrong, 2) avoid recompile on each call of `saveNamespaceRuleGroup`.

Reply:
Done in [fdb49c1](https://github.com/grafana/mimir/pull/8142/commits/fdb49c1d24b32a2f1d6e31aaa7a3427e9faf36d8)